### PR TITLE
Combustion Turbine Generator Plant Comp Refactor

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
@Myoldmopar Ready for review.

File: src/EnergyPlus/CTElectricGenerator.cc [hunk 1]
```diff
@@ -61,21 +61,21 @@
...
+#include <EnergyPlus/Plant/PlantLocation.hh>
 #include <EnergyPlus/PlantUtilities.hh>
 #include <EnergyPlus/UtilityRoutines.hh>
 
 namespace EnergyPlus {
 
 namespace CTElectricGenerator {
 
-    //__________________________________________________________________________
+    //__________________________________________________________________________;
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.cc:78]: That's a funny change. 😆

File: src/EnergyPlus/CTElectricGenerator.cc [hunk 2]
```diff
@@ -96,127 +96,59 @@ namespace CTElectricGenerator {
...
+    Array1D<CTGeneratorData> CTGenerator; // dimension to number of machines
 
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int GenNum; // Generator number counter
+    void clear_state()
+    {
+        NumCTGenerators = 0;
+        getCTInputFlag = true;
+        CTGenerator.deallocate();
+    }
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.cc:109]: All 3 global variables, check!

File: src/EnergyPlus/CTElectricGenerator.cc [hunk 3]
```diff
@@ -96,127 +96,59 @@ namespace CTElectricGenerator {
...
-                return;
-            }
-            MinCap = 0.0;
-            MaxCap = 0.0;
-            OptCap = 0.0;
-            return;
-        } // End Of InitLoopEquip
+        // empty function to emulate current behavior as of conversion to using the PlantComponent calling structure.
+        // calls from the plant side... do nothing.
+        // calls from the ElectricPowerServiceManger call the init and calculation worker functions directly.
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.cc:149]: This does beg the question whether this component really ever should've been a part of the plant system if the sim call doesn't really do anything.  But that's not a question for right now, this is looking good.

File: src/EnergyPlus/CTElectricGenerator.cc [hunk 4]
```diff
@@ -602,219 +480,213 @@ namespace CTElectricGenerator {
...
+        Real64 maxPartLoadRat = this->MaxPartLoadRat;
+
+        // Generator nominal capacity (W)
+        Real64 ratedPowerOutput = this->RatedPowerOutput;
+
+        // MAX EXHAUST FLOW PER W POWER OUTPUT COEFF
+        Real64 maxExhaustperCTPower = this->MaxExhaustperCTPower;
+
+        // design turbine inlet temperature (C)
+        Real64 designAirInletTemp = this->DesignAirInletTemp;
```
Comment (Myoldmopar) [inline src/EnergyPlus/CTElectricGenerator.cc:500]: Kinda seems like these locals aren't even really needed, but maybe there's some logic down below where it necessitates putting the values on locals.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
CI is happy, code changes look good, building and testing locally now.

**Comment by Myoldmopar** [comment]
Built and ran locally.  No issues.  A handful of BND diffs and one EIO diff which was just a reorder on the six test files including this generator.  Merging.

**Pull request merged**